### PR TITLE
Update our Code of Conduct to reflect OpenJS Foundation Code of Conduct

### PR DESCRIPTION
**Pull request by Relequestual**
Fixes json-schema-org/community#163

Moved modifications to the Contributor Covenant Code of Conduct to their own sections.
Added details about the OpenJS Foundation Code of Conduct.
Specified that the IETF BCP 54 are taken a guide only.

### CODE REVIEW

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -180,6 +182,19 @@ at [https://www.contributor-covenant.org/translations][translations].
...
+
+## Privacy Expectations
+
+As per Code of Conduct requirements set out by the OpenJS Foundation, recipients of reports will maintain the confidentiality with regard to the reporter of an incident.
+For the purposes of tracking between Code of Conduct review members, incidents may be tracked in a private organizational GitHub Repository.
+In addition to any Code of Conduct review team, GitHub organizational owners will have access to view details of reports by way of having full GitHub organizational admin access.
+
+## Escalate an issue
+
+The OpenJS Foundation provides an [escalation path](https://github.com/openjs-foundation/cross-project-council/blob/HEAD/CODE_OF_CONDUCT.md#escalation) should you feel your report has not been handled appropriatly. Recipients of reports commit to participate in the defined path of escalation when required, as required by the OpenJS Foundation Code of Conduct.
```
Comment (Relequestual) [inline CODE_OF_CONDUCT.md:198]:
```suggestion
The OpenJS Foundation provides an [escalation path](https://github.com/openjs-foundation/cross-project-council/blob/HEAD/CODE_OF_CONDUCT.md#escalate-an-issue) should you feel your report has not been handled appropriatly. Recipients of reports commit to participate in the defined path of escalation when required, as required by the OpenJS Foundation Code of Conduct.
```

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -180,6 +182,19 @@ at [https://www.contributor-covenant.org/translations][translations].
...
+
+## Escalate an issue
+
+The OpenJS Foundation provides an [escalation path](https://github.com/openjs-foundation/cross-project-council/blob/HEAD/CODE_OF_CONDUCT.md#escalate-an-issue) should you feel your report has not been handled appropriatly. Recipients of reports commit to participate in the defined path of escalation when required, as required by the OpenJS Foundation Code of Conduct.
 
-To see the deviations from the Contributor Covenant Code of Conduct version 2.1, please check the git history for this file, or manually create a diff.
\ No newline at end of file
+> The OpenJS Foundation maintains a [Code of Conduct Panel (CoCP)](https://github.com/openjs-foundation/cross-project-council/blob/HEAD/FOUNDATION_CODE_OF_CONDUCT_REQUIREMENTS.md#code-of-conduct-panel). This is a foundation-wide team established to manage escalation when a reporter believes that a report to a member project or the CPC has not been properly handled. In order to escalate to the CoCP send an email to coc-escalation@lists.openjsf.org.
```
[the remark this answers is not in the document]
Comment (Relequestual) [reply CODE_OF_CONDUCT.md:200]:
```suggestion
> The OpenJS Foundation maintains a [Code of Conduct Panel (CoCP)](https://github.com/openjs-foundation/cross-project-council/blob/HEAD/conduct/FOUNDATION_CODE_OF_CONDUCT_REQUIREMENTS.md#code-of-conduct-panel). This is a foundation-wide team established to manage escalation when a reporter believes that a report to a member project or the CPC has not been properly handled. In order to escalate to the CoCP send an email to coc-escalation@lists.openjsf.org.
```

**Review by Julian — APPROVED** [review]

**Comment by Julian** [comment]
We probably need to update a bunch of badges to reflect this kind of thing, right? Right now all the repos just mention Contributor Covenant.

**Comment by Relequestual** [comment]
Well, we subscribe to the CoC proper in full, but define a few extras.
The badges link to the CoC file in this repo.

What do you think we need to do?  (genuine question)
Modify the badge? Add more badge/s? Something else?

**Comment by Julian** [comment]
It's I guess not the worst thing in the world to leave as-is. Today yeah there's no practical difference though maybe someone who sees the badge and is familiar with the contributor covenant will at some point be surprised we in fact expect them to comply with more than that.

Perhaps an alternative is just a more generic badge, e.g.:

[Contributor Covenant](https://github.com/json-schema-org/.github/blob/main/CODE_OF_CONDUCT.md)

But I don't feel strongly if we're lazy and leave it.

**Comment by Relequestual** [comment]
I'd be fine with that and maybe suggest in addition "(CC 2.1+)". Although I don't know if that has the connotation I'd want it to.

**Review by gregsdennis — APPROVED** [review]

File: CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -1,7 +1,21 @@
...
+
+We reference the IETF BCP 54 "IETF Guidelines for Conduct" (RFC7154) as a guideline.
+We do not subscribe to its defined Reporting Transgressions of the Guidelines (Appendix A),
+as JSON Schema is not part of any IETF working group.
+
+We subscribe to the OpenJS Foundation Code of Conduct in full. The OpenJS Foundation Code of Conduct
+consists of the Contributor Covenant Code of Conduct, commitments relating to reports,
+and an escalation process.
 
 # IETF Guidelines for Conduct
```
Comment (benjagm) [inline CODE_OF_CONDUCT.md:20]: I'll suggest to remove entirely this section: "IETF Guidelines for Conduct"

**Review by benjagm — CHANGES_REQUESTED** [review]
I left a comment regarding the section: IETF Guidelines for Conduct

File: CODE_OF_CONDUCT.md (hunk 3 above)
Comment (Relequestual) [reply CODE_OF_CONDUCT.md:20]: We agreed not to remove it, but to modify it, on our previous call: https://github.com/json-schema-org/community/issues/429#issuecomment-1625052959

Comment (Relequestual) [reply CODE_OF_CONDUCT.md:20]: Updated. Please let me know if this is now clearer 🙏

Comment (benjagm) [reply CODE_OF_CONDUCT.md:20]: Honestly speaking I still find very confusing dedicating a full section of our CoC to IETF ways of working when we are no longer members of the IETF. It appears to me like we are reluctant to move forward with our CoC evolution as part of the OpenJS Foundation.

This is exactly the same thinking I shared in the OCWM but It seems I am the only one with this thinking, therefore I am approving the PR.

**Review by benjagm — APPROVED** [review]
Honestly speaking I still find very confusing dedicating a full opening section of our CoC to the IETF ways of working when we are no longer members of the IETF. It appears to me like we are reluctant to move forward with our CoC evolution as part of the OpenJS Foundation. I think 99% our Community dont know what is the IETF BCP 54 or RFC7154 because there were not part of the spec editing discussions.

I suggest to move this whole section at the bottom of the CoC after "The OpenJS Foundation Code of Conduct"

This is the same thinking I shared in the OCWM but It seems I am the only one with this thinking, therefore I am approving the PR to move forward.

File: CODE_OF_CONDUCT.md [hunk 4]
```diff
@@ -1,17 +1,34 @@
...
```
Comment (benjagm) [inline CODE_OF_CONDUCT.md:20]:
```suggestion
# Guidelines for Conduct
```

**Pull request merged**